android/app/src/main/java/org/fossasia/openevent/fragments/FeedFragment.java

```diff
@@ -157,6 +166,24 @@ public void networkUnavailable() {
 
     }
 
+    private void showProgressBar(boolean show) {
+        if(show)
+            downloadProgressDialog.show();
+        else
+            downloadProgressDialog.dismiss();
+    }
+
+    private void setupProgressBar() {
+        downloadProgressDialog = new ProgressDialog(getContext());
+        downloadProgressDialog.setIndeterminate(true);
+        downloadProgressDialog.setProgressPercentFormat(null);
+        downloadProgressDialog.setProgressNumberFormat(null);
+        downloadProgressDialog.setProgressStyle(ProgressDialog.STYLE_HORIZONTAL);
+        downloadProgressDialog.setCancelable(false);
+        String shownMessage = String.format(getString(R.string.downloading_format), getString(R.string.menu_feed));
+        downloadProgressDialog.setMessage(shownMessage);
```

Comment:
sorry for not following the thread. @iamareebjamal is it possible to make a generic class which can handle both scenarios or should we stick to this for now?